### ISSUE TITLE
docker_container - missing --cpus=<value>

### ISSUE THREAD

**Issue by benbro** [ansible/ansible#34320]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
[docker_container](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_container.py)

##### ANSIBLE VERSION
```
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/me/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 20 2017, 18:23:56) [GCC 5.4.0 20160609]
```

##### SUMMARY
Ansible is missing support for the docker run [--cpus=<value>](https://docs.docker.com/engine/admin/resource_constraints/#cpu). This flag is available since Docker 1.3.

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/docker/docker_container.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_container.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @chouseknecht @cove @dusdanig @jctanner @joshuaconner @kassiansun @softzilla @zfil
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by maxamillion** [comment]
!needs_triage

**Comment by ko-christ** [comment]
This option ~~doesn't seem to be supported~~ is `nano_cpus` in the `docker.py` API.
https://docker-py.readthedocs.io/en/stable/containers.html
https://github.com/docker/docker-py/issues/1920

**Comment by Akasurde** [comment]
resolved_by_pr #38063

**Comment by Akasurde** [comment]
@benbro Could you please check if this PR works for you ? Let us know.

**Comment by benbro** [comment]
@Akasurde this issue is about --cpus=value config but the PR use nano_cpus which I don't understand.

**Comment by Akasurde** [comment]
@benbro CPU quota is called as `nano_cpus` in API library - please refer this https://github.com/docker/docker-py/issues/1920#issuecomment-367399139

**Comment by benbro** [comment]
@Akasurde There are different ways to limit CPU in docker with the command line. This [blog](https://blog.docker.com/2017/01/cpu-management-docker-1-13/) says:
> Specifying cpu usage in nanoseconds for a user is sometimes hard to determine when you want to do simple tasks such as limiting a container to one core. 

nano_cpus is very confusing:
>  NanoCPUs - CPU quota in units of 10-9 CPUs.

Any reason not to give the user just: --cpus=value?

**Comment by Akasurde** [comment]
Agree. nano_cpus is very confusing. Maybe you can add comment in PR itself, stating that `nano_cpus` should be renamed to something less confusing. Also, to add more documentation around that parameter.

**Comment by benbro** [comment]
Already done.

**Comment by alandotcom** [comment]
What's the status on this? I'd be happy to help with any work to get things moving along

**Comment by felixfontein** [comment]
There's a PR for this (#38063), which seems to be stuck. Also, there's disagreement how the interface for this should look like.

**Comment by ansibot** [comment]
cc @akshay196 @danihodovic @dariko @jwitko @tbouvet
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @DBendit
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @WojciechowskiPiotr
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by hdiass** [comment]
Is there any update on this option to be available on docker_container module ?
thanks,

**Comment by felixfontein** [comment]
@hdiass so far, nothing happened. I'll work on a PR today; I would be glad if you could test it once it is ready.

**Comment by felixfontein** [comment]
resolved_by_pr #64683
